geode-core/src/main/java/org/apache/geode/internal/statistics/StatArchiveWriter.java

```diff
@@ -508,6 +508,9 @@ public void sampled(long nanosTimeStamp, List<ResourceInstance> resourceInstance
       }
       writeTimeStamp(nanosTimeStamp);
       for (ResourceInstance ri : resourceInstances) {
+        if (!ri.getStatValuesNotified()) {
+          ri.setStatValuesNotified(true);
+        }
```

Comment:
I think that the `if` is unnecessary here, since it's fine to call `setStatValuesNotified(true)` if the value is already true.